src/main/java/org/isf/medicalstock/manager/MovStockInsertingManager.java

```diff
@@ -418,4 +425,81 @@ private List<Movement> prepareDishargingMovement(Movement movement, boolean chec
 			return dischargeMovement;
 		}
 	}
+	
+	/**
+	 * Prepare the insert of the specified {@link Movement}
+	 *
+	 * @param movement - the movement to store.
+	 * @return {@code true} if is already used,{@code false} otherwise.
+	 * @throws OHServiceException.
+	 */
+	@Transactional(rollbackFor = OHServiceException.class)
+	public boolean prepareDishargingOrChargeMovementInventory(Movement movement) throws OHServiceException {
```

Comment:
"prepare" movement should be something to prepare an object to be persisted afterward, not to check and return a boolean, and the javadoc seems not reflecting the behaviour either (`true` if is already used?)

please what is the logic behind and what are you trying to achieve?

Reply:
The logic  behind this method is:

- to store the new movement
- to update in_quantity of Medical or MedicalWard if the movement is charge or update out_quantityof Medical or MedicalWard if the movement is discharge.

Reply:
> to store the new movement

that's why I'm saying, why to call it "prepare"? Moreover a "store" operation, like save or persist or update, should return the stored object as from JPA specs, not a boolean. 

Anyway why it is called prepare**DishargingOrCharge**MovementInventory while inside we only do `ioOperations.prepareDischargingMovement(movement)`?

And the `updateStockQuantity` and `updateMedicalWardQuantity` look copied and paste from the `MedicalStockIoOperations`, why this code duplication?